File: src/r/pe_0004.r
itos <- function(n) {
    unlist(strsplit(paste(n), NULL))
}

is_palindromic <- function(n) {
    v <- itos(n)
    identical(v, rev(v))
}

prods <- NULL
for (a in 100:999) {
    prods <- c(prods, a * (a:999))
}
prods <- unique(prods)

index <- sapply(prods, is_palindromic)
max(prods[index])

File: src/pl/pe_0004.pl
sub is_palindromic {
    $n = shift;
    $m = reverse $n;
    return $n eq $m;
}

$max = 0;

for ($a = 1; $a < 1000; $a += 1) {
    for ($b = $a; $b < 1000; $b += 1) {
        $p = $a * $b;
        if (is_palindromic($p) && $max < $p) {
            $max = $p;
        }
    }
}

print $max . "\n";
